extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -910,6 +911,29 @@ export class Bind {
   }
 
   /**
+   * Print out the current state in the console.
+   * @private
+   */
+  printAmpState_() {
+    const seen = [];
+    const seenNames = [];
+    const s = JSON.stringify(this.scope_, (key, value) => {
+      if (isObject(value)) {
+        const index = seen.indexOf(value);
+        if (index !== -1) {
+          const name = seenNames[index];
+          return `**Circular reference to '${name}'**`;
```

Comment:
Let's use node's handling of circular references:
```
> var a = {};
undefined
> a.a = a;
{ a: [Circular] }
```
You can remove `seenNames`.